llvm/lib/Support/RISCVISAInfo.cpp

```diff
@@ -715,6 +815,8 @@ RISCVISAInfo::parseArchString(StringRef Arch, bool EnableExperimentalExtension,
 
   unsigned XLen = HasRV64 ? 64 : 32;
   std::unique_ptr<RISCVISAInfo> ISAInfo(new RISCVISAInfo(XLen));
+  SmallVector<std::string, 8> SeenExts;
```

Comment:
Maybe SeenExts should be a SetVector to make it more efficient to check for duplicates?

Or maybe a MapVector storing the version as the value so we don't need ExtsVersion?

Reply:
Use MapVector instead of SmallVector